packages/caliper-fisco-bcos/lib/fiscoBcos.js

```diff
@@ -0,0 +1,143 @@
+        super(config_path);
+        this.bcType = 'fisco-bcos';
+        this.workspaceRoot = workspace_root;
+    }
+
+    /**
+     * Initialize the {FISCO BCOS} object.
+     * @async
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async init() {
+        return Promise.resolve();
+    }
+
+    /**
+     * Deploy the smart contract specified in the network configuration file to all nodes.
+     * @async
+     */
+    async installSmartContract() {
+        const fiscoBcosSettings = CaliperUtils.parseYaml(this.configPath)['fisco-bcos'];
+        try {
+            await installSmartContractImpl.run(fiscoBcosSettings, this.workspaceRoot);
+        } catch (error) {
+            commLogger.error(Color.error(`FISCO BCOS smart contract install failed: ${(error.stack ? error.stack : error)}`));
+            throw error;
+        }
+    }
+
+    /**
+     * Get a context for subsequent operations
+     * 'engine' attribute of returned context object must be reserved for benchmark engine to extend the context
+     *  engine = {
+     *   submitCallback: callback which must be called once new transaction(s) is submitted, it receives a number argument which tells how many transactions are submitted
+     * }
+     * @param {String} name name of the context
+     * @param {Object} args adapter specific arguments
+     * @param {Integer} clientIdx the client index
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async getContext(name, args, clientIdx) {
+        return Promise.resolve();
+    }
+
+    /**
+     * Release a context as well as related resources
+     * @param {Object} context adapter specific object
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async releaseContext(context) {
+        return Promise.resolve();
+    }
+
+    /**
+     * Invoke the given smart contract according to the specified options. Multiple transactions will be generated according to the length of args.
+     * @param {object} context The FISCO BCOS context returned by {getContext}.
+     * @param {string} contractID The name of the smart contract.
+     * @param {string} contractVer The version of the smart contract.
+     * @param {Array} args Array of JSON formatted arguments for transaction(s). Each element contains arguments (including the function name) passing to the smart contract. JSON attribute named transaction_type is used by default to specify the function name. If the attribute does not exist, the first attribute will be used as the function name.
+     * @param {number} timeout The timeout to set for the execution in seconds.
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async invokeSmartContract(context, contractID, contractVer, args, timeout) {
+        const fiscoBcosSettings = CaliperUtils.parseYaml(this.configPath)['fisco-bcos'];
```

Comment:
Is it required to parse this here, or could it be cached? The `invokeSmartContract` is a hot path, so if repeated parsing can be avoided, it will be more efficient

Reply:
Solved :)